metaflow/plugins/uv/bootstrap.py

```diff
@@ -0,0 +1,86 @@
+import os
+import subprocess
+import sys
+
+from metaflow.util import which
+
+if __name__ == "__main__":
+
+    def run_cmd(cmd, stdin_str=None):
+        result = subprocess.run(
+            cmd,
+            shell=True,
+            input=stdin_str,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        if result.returncode != 0:
+            print(f"Bootstrap failed while executing: {cmd}")
+            print("Stdout:", result.stdout)
+            print("Stderr:", result.stderr)
+            sys.exit(1)
+
+    def install_uv():
+        if which("uv"):
+            return
+
+        print("Installing uv...")
+        uv_version = "0.1.8"
+        cmd = f"""set -e;
+            curl -LsSf https://astral.sh/uv/install.sh | env UV_UNMANAGED_INSTALL="/uv_install" UV_VERSION="{uv_version}" sh
+            """
+        run_cmd(cmd)
+
+        uv_path = "/uv_install"
+        if os.path.isdir(uv_path):
+            os.environ["PATH"] += os.pathsep + uv_path
+        else:
+            print("uv failed to install.")
+            print(os.listdir("/uv_install"))
+
+    def get_dependencies(datastore_type):
+        datastore_packages = {
+            "s3": ["boto3"],
+            "azure": [
+                "azure-identity",
+                "azure-storage-blob",
+                "azure-keyvault-secrets",
+                "simple-azure-blob-downloader",
+            ],
+            "gs": [
+                "google-cloud-storage",
+                "google-auth",
+                "simple-gcp-object-downloader",
+                "google-cloud-secret-manager",
+            ],
+        }
+
+        if datastore_type not in datastore_packages:
+            raise NotImplementedError(
+                "Unknown datastore type: {}".format(datastore_type)
+            )
+
+        return datastore_packages[datastore_type] + ["requests"]
+
+    def sync_uv_project(datastore_type):
+        print("Syncing uv project...")
+        dependencies = " ".join(get_dependencies(datastore_type))
+        cmd = f"""set -e;
+            ls -la;
+            uv sync --frozen --no-install-package metaflow;
+            uv pip install {dependencies} --strict
```

Comment:
I am a little confused here: why are the dependencies added here. This makes it non-reproducible (since the packages could change). Could we instead make it so that these dependencies are included before packaging the lock and pyproject.toml files?

Reply:
We wouldn't want to modify the lock/pyproject files since the user controls those files and can lead to surprises for them. In most cases, this step will be a no-op, and the expectation always with the dependencies we introduce is that those are safe and validated dependencies.

Reply:
I agree to not modifying the user level lock file but I feel we could modify it *before* packaging it. I know this is not the same as other decorators but this definitely breaks the "your environment won't change" and is equivalent to doing a "pip install" at runtime which was one of the drivers for the initial conda environment.

Reply:
we can chat more in person - but the user-specific env actually doesn't change here, so I don't see where the _definitely breaks_ is coming from